packages/react-instantsearch-core/src/core/createInstantSearchManager.js

```diff
@@ -328,7 +328,7 @@ export default function createInstantSearchManager({
     // - Algoliasearch API Client < v4 with cache disabled
     // - Third party clients (detected by the `addAlgoliaAgent` function missing)
     if (
-      !client.transporter &&
+      (!client.transporter || client._cacheHydrated) &&
```

Comment:
Don't we want to bypass all the other checks if `client._cacheHydrated` is `true`?